Add coverage to regex_check method

szuananwar opened the pull request.

szuananwar committed:
Add test_regex_check method in test_builders.py to text regex_check method

test_regex is in tests/builder/test_builders.py
regex_check method is in buildesystem/checks.py

szuananwar committed:
Added symbolic link for status_regex.yml in tests/builders

szuananwar committed:
Added buildspec for stderr in status_regex.yml file

szuananwar committed:
Added .idea/ to gitignore

szuananwar committed:
Run style check and fixed some long >80 column

shahzebsiddiqui commented on tests/builders/status_regex.yml:
```diff
@@ -0,0 +1 @@
+/Users/anwar/Documents/GitHub/buildtest/tutorials/test_status/status_regex.yml
```
this symbolic link looks hardcoded to your filesystem this wont work. Please use relative paths

szuananwar replied:
> this symbolic link looks hardcoded to your filesystem this wont work. Please use relative paths

Done, thanks!

shahzebsiddiqui commented on tutorials/test_status/status_regex.yml:
```diff
@@ -19,4 +19,26 @@ buildspecs:
-        exp: "^(123FAIL)$"
\ No newline at end of file
+        exp: "^(123FAIL)$"
+
+  status_regex_stderr_pass:
+    executor: generic.local.bash
+    type: script
+    tags: [ system ]
+    description: Pass test based on regular expression
+    run: echo "PASS"
+    status:
+      regex:
+        stream: stderr
+        exp: "^(PASS)$"
+
+  status_regex_stderr_fail:
+    executor: generic.local.bash
+    type: script
+    tags: [ system ]
+    description: Pass test based on regular expression
+    run: echo "FAIL"
+    status:
+      regex:
+        stream: stderr
+        exp: "^(123FAIL)$"
```
since you added 2 more tests here you will want to update the documentation in particular https://buildtest.readthedocs.io/en/devel/buildspecs/buildspec_overview.html#passing-test-based-on-regular-expression page where you should update the paragraph to explain what this test is doing. You should emphasize the important lines.

szuananwar committed:
Update status_regex.yml

shahzebsiddiqui reviewed (changes requested):
This look better but a few more changes are needed in order to merge this. Please see my comments.

Mendi03 reviewed (commented):
@shahzebsiddiqui Please see my comment.

Mendi03 commented on tests/builders/test_builders.py:
```diff
@@ -138,3 +138,14 @@ def test_file_count():
     cmd.build()
+
+
+def test_regex_check():
+    """This test buildspec using status check with  'status_regex'"""
+    cmd = BuildTest(
+        buildspecs=[
+            os.path.join(here, "status_regex.yml")],
+        buildtest_system=system,
+        configuration=config,
+    )
+    cmd.build()
```
@shahzebsiddiqui  When I tried running coverage, I see the error `There are no config files to process.`. This might be because on my side it says that the symbolic link is invalid. However, @szuananwar says that the symbolic link works on her side and the regresion test works properly. Is there a reason why it works for her but not during the regresion test by github?

szuananwar committed:
Merge branch 'buildtesters:devel' into test_regex_check

szuananwar committed:
Updated the paragraph in https://buildtest.readthedocs.io/en/devel/buildspecs/buildspec_overview.html#passing-test-based-on-regular-expression
 to include error file pass and fail state and important lines has been emphasized.

shahzebsiddiqui committed:
fix issue with symbolic link
